fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java

```diff
@@ -96,5 +106,80 @@ public Void visitExecuteAsStatement(ExecuteAsStmt stmt, ConnectContext session)
+                    if (stmt.isPasswordPlain()) {
+                        // convert plain password to scramble
+                        scramblePassword = MysqlPassword.makeScrambledPassword(stmt.getOriginalPassword());
+                    } else {
+                        scramblePassword = MysqlPassword.checkPassword(stmt.getOriginalPassword());
+                    }
+                } catch (DdlException | AnalysisException e) {
+                    // TODO AnalysisException used to raise in all old methods is captured and translated to SemanticException
+                    // that is permitted to throw during analyzing phrase under the new framework for compatibility.
+                    // Remove it after all old methods migrate to the new framework
+                    throw new SemanticException(e.getMessage());
+                }
+            } else {
+                scramblePassword = new byte[0];
+            }
+            stmt.setScramblePassword(scramblePassword);
+
+            /*
+             * IDENTIFIED WITH
+             */
+            if (!Strings.isNullOrEmpty(stmt.getAuthPlugin())) {
+                stmt.setAuthPlugin(stmt.getAuthPlugin().toUpperCase());
+                if (AuthPlugin.AUTHENTICATION_LDAP_SIMPLE.name().equals(stmt.getAuthPlugin())) {
+                    stmt.setUserForAuthPlugin(stmt.getAuthString());
+                } else if (AuthPlugin.MYSQL_NATIVE_PASSWORD.name().equals(stmt.getAuthPlugin())) {
+                    // in this case, authString is password
+                    // convert password to hashed password
+                    if (!Strings.isNullOrEmpty(stmt.getAuthString())) {
+                        if (stmt.isPasswordPlain()) {
+                            // convert plain password to scramble
+                            scramblePassword = MysqlPassword.makeScrambledPassword(stmt.getAuthString());
+                        } else {
+                            try {
+                                scramblePassword = MysqlPassword.checkPassword(stmt.getAuthString());
+                            } catch (AnalysisException e) {
+                                // TODO AnalysisException used to raise in all old methods is captured and translated to SemanticException
+                                // that is permitted to throw during analyzing phrase under the new framework for compatibility.
+                                // Remove it after all old methods migrate to the new framework
+                                throw new SemanticException(e.getMessage());
+                            }
+                        }
+                    } else {
+                        scramblePassword = new byte[0];
+                    }
+                    stmt.setScramblePassword(scramblePassword);
+                } else if (AuthPlugin.AUTHENTICATION_KERBEROS.name().equalsIgnoreCase(stmt.getAuthPlugin()) &&
```

Comment:
I suggest you use `equals()` instead of `equalsIgnoreCase()` since `authPlugin` already converted to upper case in line 144.